tables/atom.py

```diff
@@ -652,29 +675,29 @@ def _create_numeric_class(baseclass, itemsize):
     classdict = {'itemsize': itemsize, 'type': type_,
                  '__doc__': "Defines an atom of type ``%s``." % type_}
 
-    def __init__(self, shape=(), dflt=baseclass._defvalue):
+    def __init__(self, shape: Shape=(), dflt: Any=baseclass._defvalue):
         Atom.__init__(self, self.type, shape, dflt)
     classdict['__init__'] = __init__
     return type('%sAtom' % prefix, (baseclass,), classdict)
 
 
-Int8Atom = _create_numeric_class(IntAtom, 1)
-Int16Atom = _create_numeric_class(IntAtom, 2)
-Int32Atom = _create_numeric_class(IntAtom, 4)
-Int64Atom = _create_numeric_class(IntAtom, 8)
-UInt8Atom = _create_numeric_class(UIntAtom, 1)
-UInt16Atom = _create_numeric_class(UIntAtom, 2)
-UInt32Atom = _create_numeric_class(UIntAtom, 4)
-UInt64Atom = _create_numeric_class(UIntAtom, 8)
+Int8Atom : type[IntAtom] = _create_numeric_class(IntAtom, 1)
```

Comment:
I'm not especially fond of this, and I'm not sure that the typing is 'correct'. An option would be to simply expand this and create the classes vebatim. I think that would probably be more elegant for the type system, but would of course require registering them in some other way.